test/Conversion/TorchToTcp/misc.mlir

```diff
@@ -322,3 +322,22 @@ func.func @torch.aten.size.int(%arg0: !torch.vtensor<[?,?],f32>) -> () {
   %0 = torch.aten.size.int %arg0, %int0 : !torch.vtensor<[?,?],f32>, !torch.int -> !torch.int
   return
 }
+
+// -----
+
+// CHECK-LABEL:  @torch.aten.broadcast_to(
+// CHECK-SAME:   %[[ARG:.*]]: !torch.vtensor<[1,2,1,2],f32>) {
+// CHECK:        %[[TENSOR:.*]] = torch_c.to_builtin_tensor %[[ARG]] : !torch.vtensor<[1,2,1,2],f32> -> tensor<1x2x1x2xf32>
+// CHECK:        %[[CONSTANT:.*]] = torch.constant.int 4
+// CHECK:        %[[CAST0:.*]] = torch_c.to_i64 %[[CONSTANT]]
+// CHECK:        %[[BROADCAST_DIM0:.*]] = arith.index_cast %[[CAST0]] : i64 to index
+// CHECK:        %[[CAST1:.*]] = torch_c.to_i64 %[[CONSTANT]]
+// CHECK:        %[[BROADCAST_DIM1:.*]] = arith.index_cast %[[CAST1]] : i64 to index
+// CHECK:        %{{.*}} = tcp.broadcast %[[TENSOR]], %[[BROADCAST_DIM0]], %[[BROADCAST_DIM1]] {axes = [0, 2]} : tensor<1x2x1x2xf32>, index, index -> tensor<4x2x4x2xf32>
+func.func @torch.aten.broadcast_to(%arg0: !torch.vtensor<[1,2,1,2],f32>) -> () {
```

Comment:
Thanks for adding the lit test (this was missed earlier). Can you please also onboard this to the `aot_compile` macro for e2e testing? I usually just have chat-GPT write me the pytorch model to produce this Torch dialect program, and feed that into the model loader.

It gave me this:
```python
class MyModel(torch.nn.Module):
    def __init__(self):
        super(MyModel, self).__init__()

    def forward(self, arg0):
        # Define the desired shape as a list
        shape = [4, 2, 4, 2]
        # Broadcast the input tensor to the desired shape
        output = arg0.expand(*shape)
        return output

# Example usage
model = MyModel()
sample_input = torch.randn(1, 2, 1, 2)  # Example input shape [1, 2, 1, 2]
output = model(sample_input)
print(output)
```

Reply:
@AaronStGeorge I did it for you here: https://github.com/cruise-automation/mlir-tcp/pull/63 :)

Reply:
Thanks @sjain-stanford :) !